src/network/network_gui.cpp

```diff
@@ -2413,6 +2413,26 @@ struct NetworkAskRelayWindow : public Window {
 		this->InitNested(0);
 	}
 
+	void Close([[maybe_unused]] int data = 0) override
```

Comment:
not "maybe" unused. Very used :)

What I find difficult to see now, is whether `WID_NAR_YES_ONCE` for example is not also `0`. How do I know that this default value is not used?

Maybe we should use `-2` as default?

Reply:
Probably passing the widget index here is not correct. Defaulting to -1 would be a design pattern change from how OnInvalidateData is used. 

Given we also need to handle CloseWindowByClass() which should not issue ConnectFailure, and is definitely not a widget, I think it means sense to revert the last commit here and then pass handled/unhandled state instead.

Reply:
Still this pesky `maybe_unused` that is annoying me; nice solution otherwise :)

Reply:
Hmm, well, I followed the pattern used elsewhere maybe they are wrong too?

Reply:
We really should only add it in places where it is unused. Adding it to places where it is used, is weird?

The parent can still have it, and all other definitions of `Close`, but here it is clearly used? Feels weird to keep it. Before you know it, we just add it everywhere :D